### PR TITLE
[FLOSS T1] Change checking of stale issues/PRs to GitHub action

### LINKED ISSUE #4476: [FLOSS] Switch from probot/stale to github/stale
We should switch from the current Probot based "stale issue bot" to [the one provided by GitHub](https://github.com/marketplace/actions/close-stale-issues).

Not only is the Probot based one, seemingly no longer maintained (see https://github.com/probot/stale/issues/385 and https://github.com/probot/probot.github.io/pull/376), but the GitHub one also has a lot more options. Including [one option](https://github.com/actions/stale#close-issue-reason) which allows closing issues as "not planned" instead of "completed".

### PR DESCRIPTION
**Pull request by 0x6178656c**
As discussed in #4476 there is an additional step required when removing the old configuration. There @stefnotch did the research on how to carry out the swtich from probot/stale to the GitHub action.

One notable difference to the previous setup is that the check would now only run daily. (It is straightforward to change this by modifying the cron job configuration.)

Afaict, the cron job would run on a runner from GitHub.

Closes #4476

## Basics

- [x] Short descriptions of your changes are in the release notes
      (added as entry in `doc/news/_preparation_next_release.md` which
      contains `_(my name)_`)
      **Please always add something to the release notes.**
- [x] Details of what you changed are in commit messages
      (first line should have `module: short statement` syntax)
- [ ] References to issues, e.g. `close #X`, are in the commit messages.
- [ ] The buildservers are happy. If not, fix **in this order**:
  - [ ] add a line in `doc/news/_preparation_next_release.md`
  - [ ] reformat the code with `scripts/dev/reformat-all`
  - [ ] make all unit tests pass
  - [ ] fix all memleaks
- [x] The PR is rebased with current master.

## Checklist

- [ ] I added unit tests for my code
- [ ] I fully described what my PR does in the documentation
      (not in the PR description)
- [ ] I fixed all affected documentation (see [Documentation Guidelines](https://www.libelektra.org/devgettingstarted/documentation))
- [ ] I fixed all affected decisions (see [Decision Process](https://www.libelektra.org/decisions/decision-process))
- [ ] I added code comments, logging, and assertions as appropriate (see [Coding Guidelines](https://www.libelektra.org/devgettingstarted/coding))
- [ ] I updated all meta data (e.g. README.md of plugins and [METADATA.ini](https://master.libelektra.org/doc/METADATA.ini))
- [ ] I mentioned [every code](/.reuse/dep5) not directly written by me in [reuse syntax](https://reuse.software/)

## Review

- [ ] Documentation is introductory, concise, good to read and describes everything what the PR does
- [ ] Examples are well chosen and understandable
- [ ] Code is conforming to [our Coding Guidelines](https://master.libelektra.org/doc/CODING.md)
- [ ] APIs are conforming to [our Design Guidelines](https://master.libelektra.org/doc/DESIGN.md)
- [ ] Code is consistent to [our Design Decisions](https://master.libelektra.org/doc/decisions)

## Labels

- [ ] Add the "work in progress" label if you do not want the PR to be reviewed yet.
- [ ] Add the "ready to merge" label **if the basics are fulfilled** and no further pushes are planned by you.

### CODE REVIEW

**Comment by flo91** [comment]
@markus2330 Are you ok with that?

> One notable difference to the previous setup is that the check would now only run daily.

If so, then please do the switch as you announced in https://github.com/ElektraInitiative/libelektra/issues/4476#issuecomment-1285128955

File: .github/workflows/stale.yml [hunk 1]
```diff
@@ -0,0 +1,49 @@
+name: Close inactive issues and PRs
+on:
+  schedule:
+    - cron: "30 1 * * *"
```
Comment (markus2330) [inline .github/workflows/stale.yml:4]: Which timezone is this?

File: .github/workflows/stale.yml [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+jobs:
+  close-issues:
+    runs-on: ubuntu-latest
+    permissions:
+      issues: write
+      pull-requests: write
+    steps:
+      - uses: actions/stale@v5
```
Comment (markus2330) [inline .github/workflows/stale.yml:13]: Any specific reason why you didn't use the latest version (v6.0.1)?

**Review by markus2330 — COMMENTED** [review]
LGTM, let us see if it works.

File: .github/workflows/stale.yml (hunk 1 above)
Comment (0x6178656c) [reply .github/workflows/stale.yml:4]: I fear this is undefined when using GitHub runners.

There is an issue where others have requested the possibility of setting a timezone: https://github.com/actions/runner/issues/1423

If this is important, I could do further research.

File: .github/workflows/stale.yml (hunk 2 above)
Comment (0x6178656c) [reply .github/workflows/stale.yml:13]: No reason. I adopted the code from <https://docs.github.com/en/actions/managing-issues-and-pull-requests/closing-inactive-issues>.

File: .github/workflows/stale.yml (hunk 1 above)
Comment (markus2330) [reply .github/workflows/stale.yml:4]: No, not important.

**Pull request merged**